gpytorch/lazy/constant_mul_lazy_variable.py

```diff
@@ -10,9 +10,11 @@
 class ConstantMulLazyVariable(LazyVariable):
     def __init__(self, lazy_var, constant):
         if torch.is_tensor(constant):
-            constant = constant
+            if constant.numel() == 1:
+                constant = constant.squeeze()
         else:
-            constant = torch.tensor([constant], device=lazy_var.device, dtype=torch.float32)
+            constant = torch.tensor(constant, device=lazy_var.device, dtype=torch.float32)
```

Comment:
This doesn't create a scalar - it creates a 1D tensor of size `constant` (expects constant to be an int, not a float)

Reply:
In addition, we probably shouldn't be constraining the `dtype` to be a float - we probably want it to have the same `dype` as the lazy variable (which could be a float or a double).

Reply:
What do you mean? `torch.tensor(3.2)` makes a scalar

Reply:
Whoops my b

Reply:
I suggest we give LazyVariable a `dtype` property that we can use in situations like this. What do you think?